File: DS-p1-ejercicio3-libre/src/Director.py
class Director:
    def __init__(self, builder):
        self._builder = builder

    def builder_car(self):
        self._builder.create_new_car()
        self._builder.add_autonomia()
        self._builder.add_fuel_type()
        self._builder.add_coste_recarga()
        self._builder.add_tiempo_recarga()

File: lib/director.dart
import 'builder.dart';

class Director {
  final CarBuilder _builder;

  Director(this._builder);

  void construir() {
    _builder.addModelo();
    _builder.addTipoCombustible();
    _builder.addAutonomia();

    if (_builder is BuilderElectrico || _builder is BuilderHibrido) {
      _builder.addTiempoRecarga();
      _builder.addCosteRecarga();
    }
  }
}
